Validation/MtdValidation/plugins/BtlDigiHitsValidation.cc

```diff
@@ -0,0 +1,279 @@
+#include "Geometry/MTDNumberingBuilder/interface/MTDTopology.h"
+
+#include "Geometry/MTDGeometryBuilder/interface/ProxyMTDTopology.h"
+#include "Geometry/MTDGeometryBuilder/interface/RectangularMTDTopology.h"
+
+
+class BtlDigiHitsValidation : public DQMEDAnalyzer {
+
+public:
+  explicit BtlDigiHitsValidation(const edm::ParameterSet&);
+  ~BtlDigiHitsValidation() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+
+private:
+  void bookHistograms(DQMStore::IBooker &,
+		      edm::Run const&,
+		      edm::EventSetup const&) override;
+
+  void analyze(const edm::Event&, const edm::EventSetup&) override;
+
+  // ------------ member data ------------
+
+  const MTDGeometry* geom_;
+  const MTDTopology* topo_;
+
+  const std::string folder_;
+  const std::string infoLabel_;
+  const std::string btlDigisCollection_;
+
+  int eventCount_;
+
+  edm::EDGetTokenT<BTLDigiCollection> btlDigiHitsToken_;
+
+  // --- histograms declaration
+
+  MonitorElement* meNhits_[2];
+
+  MonitorElement* meHitCharge_[2];
+  MonitorElement* meHitTime_[2];
+
+  MonitorElement* meOccupancy_[2];
+
+  MonitorElement* meHitX_[2];
+  MonitorElement* meHitY_[2];
+  MonitorElement* meHitZ_[2];
+  MonitorElement* meHitPhi_[2];
+  MonitorElement* meHitEta_[2];
+
+  MonitorElement* meHitTvsQ_[2];
+  MonitorElement* meHitQvsPhi_[2];
+  MonitorElement* meHitQvsEta_[2];
+  MonitorElement* meHitQvsZ_[2];
+  MonitorElement* meHitTvsPhi_[2];
+  MonitorElement* meHitTvsEta_[2];
+  MonitorElement* meHitTvsZ_[2];
+
+};
+
+
+// ------------ constructor and destructor --------------
+BtlDigiHitsValidation::BtlDigiHitsValidation(const edm::ParameterSet& iConfig):
+  geom_(nullptr),
+  topo_(nullptr),
+  folder_(iConfig.getParameter<std::string>("folder")),
+  infoLabel_(iConfig.getParameter<std::string>("moduleLabel")),
+  btlDigisCollection_(iConfig.getParameter<std::string>("btlDigiHitsCollection")),
+  eventCount_(0) {
+
+  btlDigiHitsToken_ = consumes <BTLDigiCollection> (edm::InputTag(std::string(infoLabel_),
+								  std::string(btlDigisCollection_)));
+
+}
+
+BtlDigiHitsValidation::~BtlDigiHitsValidation() {
+}
+
+
+// ------------ method called for each event  ------------
+void BtlDigiHitsValidation::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+
+  using namespace edm;
+
+  edm::LogInfo("EventInfo") << " Run = " << iEvent.id().run() << " Event = " << iEvent.id().event();
+
+  edm::ESHandle<MTDGeometry> geometryHandle;
+  iSetup.get<MTDDigiGeometryRecord>().get(geometryHandle);
+  geom_ = geometryHandle.product();
+
+  edm::ESHandle<MTDTopology> topologyHandle;
+  iSetup.get<MTDTopologyRcd>().get(topologyHandle);
+  topo_ = topologyHandle.product();
+
+  edm::Handle<BTLDigiCollection> btlDigiHitsHandle;
+  iEvent.getByToken(btlDigiHitsToken_, btlDigiHitsHandle);
+
+  if( ! btlDigiHitsHandle.isValid() ) {
+    edm::LogWarning("DataNotFound") << "No BTL DIGI hits found";
+    return;
+  }
```

Comment:
Could this just throw on missing product? (probably not given how the phase2 validation configuration is constructed)

Reply:
@casarsa @makortel I think we could combine an appropriate choice of era and the new validHandle from https://indico.cern.ch/event/794489/contributions/3300916/attachments/1787057/2910003/New_Event_get_methods_.pdf

Reply:
Switched to validHandle